src/SwiftUI/SwiftUI/Color.cs

```diff
@@ -75,6 +76,59 @@ public Color (double hue, double saturation, double brightness, double opacity)
 		{
 			Data = CreateFromHSBO (hue, saturation, brightness, opacity);
 		}
+
+		public Color (RGBColorSpace colorSpace, double red, double green, double blue, double opacity)
+		{
+			var opaqueRBGColorspaceMetadata = SwiftType.Of (typeof (RGBColorSpace))!;
+			var result = TaggedPointer.AllocHGlobal (opaqueRBGColorspaceMetadata.NativeDataSize);
+			try {
+				Data = CreateFromRGBColorSpaceRedGreenBlueOpacity (GetSwiftUIColorSpace (colorSpace, result).Pointer, red, green, blue, opacity);
+			} catch {
+				result.Dispose ();
+				throw;
+			}
+		}
+
+		public Color (RGBColorSpace colorSpace, double white, double opacity)
+		{
+			var opaqueRBGColorspaceMetadata = SwiftType.Of (typeof (RGBColorSpace))!;
+			var result = TaggedPointer.AllocHGlobal (opaqueRBGColorspaceMetadata.NativeDataSize);
+			try {
+				Data = CreateFromRGBColorSpaceWhiteOpacity (GetSwiftUIColorSpace (colorSpace, result).Pointer, white, opacity);
+			} catch {
+				result.Dispose ();
+				throw;
+			}
+		}
+
+		static TaggedPointer GetSwiftUIColorSpace (RGBColorSpace colorSpace, TaggedPointer result)
+		{
+			return colorSpace switch
+			{
+				RGBColorSpace.DisplayP3 => ColorSpaceDisplayP3 (result),
+				RGBColorSpace.sRGB => ColorSpacesRGB (result),
+				RGBColorSpace.sRGBLinear => ColorSpacesRGBLinear (result),
+				_ => throw new NotSupportedException (),
+			};
+		}
+
+		static TaggedPointer ColorSpaceDisplayP3 (TaggedPointer  result)
```

Comment:
Set up this way as it was easier to call it from the switch. Happy to change it if you don't like the style.

Reply:
It looks like it would simplify things if we just don't use `TaggedPointer`? For these cases, we are always allocating the memory, so I don't think it buys us anything over `Marshal.AllocHGlobal`

Reply:
Done.